src/envoy/http/service_control/service_control_call_impl.cc

```diff
@@ -130,20 +130,21 @@ void ServiceControlCallImpl::createIamTokenSub() {
 
 ServiceControlCallImpl::ServiceControlCallImpl(
     FilterConfigProtoSharedPtr proto_config, const Service& config,
-    ServiceControlFilterStats& filter_stats,
+    const std::string& stats_prefix,
     Envoy::Server::Configuration::FactoryContext& context)
     : filter_config_(*proto_config),
       token_subscriber_factory_(context),
       tls_(context.threadLocal().allocateSlot()) {
   // Pass shared_ptr of proto_config to the function capture so that
   // it will not be released when the function is called.
-  tls_->set([proto_config, &config, &filter_stats,
+  tls_->set([proto_config, &config, stats_prefix, &scope = context.scope(),
```

Comment:
Will `stats_prefix` be copied in the lambda capture? Or captured by reference because the original variable is already a reference?

Reply:
it is a copy, you have to specify "&" if you want to pass in as reference.

Reply:
Right, got confused because the variable being captured is a reference. But this will copy.